gsy_framework/data_classes.py

```diff
@@ -49,15 +49,20 @@ def __init__(self, id: str, creation_time: DateTime, price: float, energy: float
         self.id = str(id)
         self.creation_time = creation_time
         self.time_slot = time_slot  # market slot of creation
-        self.original_price = original_price or price
+        self.original_price = limit_float_precision(original_price or price)
         self.price = limit_float_precision(price)
         self.energy = limit_float_precision(energy)
         self.type = self.__class__.__name__
 
     @property
     def energy_rate(self) -> float:
         """Dynamically calculate rate of energy."""
-        return limit_float_precision(self.price / self.energy)
+        return round(self.price / self.energy, 5)
```

Comment:
should be put the "5" into a constant to change it in one place if needed later?

Reply:
Good idea, let me adapt it first.